src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -122,7 +122,8 @@ private static function compilePattern(Route $route, string $pattern, bool $isHo
 
         // Match all variables enclosed in "{}" and iterate over them. But we only want to match the innermost variable
         // in case of nested "{}", e.g. {foo{bar}}. This in ensured because \w does not match "{" or "}" itself.
-        preg_match_all('#\{(!)?(\w+)\}#', $pattern, $matches, \PREG_OFFSET_CAPTURE | \PREG_SET_ORDER);
+        $routeParamsPattern = $needsUtf8 ? '#\{(!)?([\p{L}\d_]+)\}#u' : '#\{(!)?(\w+)\}#';
```

Comment:
there is no need to check got unicode support: `\pL` always works
this means that the regexp can unconditionally be: `'#\{(!)?([\w\pL]++)\}#'`

not that there are other occurrences of `\w` in this very file

Reply:
Yes, I agree. Anyway, I found that I also need to make some fixes to support utf-8 characters in regex of compiled routes.

Reply:
@nicolas-grekas Maybe it's not so bad to split the logic. 

> Matching characters by Unicode property is not fast, because PCRE has to do a multistage table lookup in order to find a character's property. That is why the traditional escape sequences such as \d and \w do not use Unicode properties in PCRE by default

Reply:
In non-unicode mode, PCRE doesn't use unicode tables.

Reply:
@nicolas-grekas in non-unicode mode \pL does not handle unicode characters too. https://www.phpliveregex.com/p/BbM

Reply:
Which is exactly what we want, isn't it? :)

Reply:
I can't get you, sorry. Maybe I am wrong but currently added test case fails with `\w\pL]++` without unicode flag too

Reply:
I mean that:
`$routeParamsPattern = $needsUtf8 ? '#\{(!)?([\p{L}\d_]+)\}#u' : '#\{(!)?(\w+)\}#';`
brings exactly the same as:
`$routeParamsPattern = '#\{(!)?([\w\pL]++)\}#';`
but it is simpler.

Reply:
@nicolas-grekas http://sandbox.onlinephpfunctions.com/code/80e7af06b8ac3ab06a30b007afd9732c4c899533

Reply:
You're missing my point, I don't know how to make it more clear... :)
Yes, I know `\pL` behaves differently with/without the `u` modifier.
That's why I'm suggesting to remove the `$needsUtf8 ? ... : ...` operator, since it duplicates a check that PCRE already does natively.

Reply:
You are suggesting to use `#\{(!)?([\w\pL]++)\}#`. Which check PCRE will do natively? Sorry for being such meticulous :)

Reply:
When the `u` modifier is set (aka when `$needsUtf8` is true, aka when the `utf8` option is set), PCRE will use Unicode tables. It will use ASCII tables otherwise.

Reply:
So we still need to conditionally set u modifier, right?

Reply:
That's already done: https://github.com/symfony/symfony/blob/4dd13ab31542b30d2fe37f9849d9a240b2de8c75/src/Symfony/Component/Routing/RouteCompiler.php#L236

Reply:
But it doesn't work. We have to use the modifier u in this regex as well. Because otherwise route params won't be parsed.

Reply:
Oh, that's the point. But it doesn't work in my cases and I am still sure that we need to add `u` on currently discussed string.
Anyway I will wait for full PR then. Thanks for the time!